core/capabilities/remote/target/client_test.go

```diff
@@ -166,14 +163,13 @@ func testClient(ctx context.Context, t *testing.T, numWorkflowPeers int, workflo
 	}
 
 	callers := make([]commoncap.TargetCapability, numWorkflowPeers)
-	srvcs := make([]services.Service, numWorkflowPeers)
+
 	for i := 0; i < numWorkflowPeers; i++ {
 		workflowPeerDispatcher := broker.NewDispatcherForNode(workflowPeers[i])
 		caller := target.NewClient(capInfo, workflowDonInfo, workflowPeerDispatcher, workflowNodeResponseTimeout, lggr)
-		require.NoError(t, caller.Start(ctx))
+		servicetest.Run(t, caller)
```

Comment:
This helper calls `Start` and immediately registers `Close` for `Cleanup`, so that even if the test never completes, we will ensure we don't leak goroutines.

Reply:
Possible to add to linter?

Reply:
I'm not sure how to make a useful rule that isn't guessing, since it's not always wrong to call Start and Close yourself.